engine/src/main/java/io/zeebe/engine/processing/common/EventHandle.java

```diff
@@ -25,52 +25,66 @@
   private final KeyGenerator keyGenerator;
   private final MutableEventScopeInstanceState eventScopeInstanceState;
 
+  // TODO (saig0): use immutable states only (#6200)
```

Comment:
FYI when working on Job.ThrowError I opted not to use this class again, I split it up into the parts that modify state and parts that don't and had to refactor the logic so much, that it didn't resemble well the existing class. Especially, because Job.ThrowError is using error event handle, so there was more logic to look at. 

Same is currently true for the work on timer processors. We pretty much inlined the methods of this class into the timer processor and then migrated the inlined version.

This may be questionable in terms of DRY, but has some benefits in terms of reducing dependencies. 

There is stuff in `JobThrowErrorProcessor` which could be harvested as part of #6200. And I guess we could make a deliberate decision whether we want to keep this class. If so, we need some refactoring in JobThrowError processor to use the migrated class again.

Reply:
I think it is okay to use this class for now because it doesn't modify the state here effectively. In the end, I would like to avoid duplication. But let's do it as part of #6200.